File: server/Bank.Core/JsonModels/Bank/Credits/CreditModel.cs
using Bank.Core.Enums;

namespace Bank.Core.JsonModels.Bank.Credits;

public class CreditModel
{
    public long Id { get; set; }
    public long CustomerId { get; set; }
    public string CustomerDisplayName { get; set; } = string.Empty;
    public CreditType CreditType { get; set; }
    public decimal GrantedAmount { get; set; }
    public int TermMonths { get; set; }
    public decimal AppliedAnnualInterestRate { get; set; }
    public decimal AppliedGrantingFee { get; set; }
    public bool CustomerWasVipAtCreation { get; set; }
    public decimal PlannedMonthlyPaymentAmount { get; set; }
    public CreditStatus Status { get; set; }
    public DateTime GrantedAtUtc { get; set; }
    public DateTime? RepaidAtUtc { get; set; }
}

File: client/src/types/JsonModels/Bank/Credits/CreditModel.ts
import type { CreditStatus } from "../../Enums/CreditStatus";
import type { CreditType } from "../../Enums/CreditType";

export interface CreditModel
{
	id: number;
	customerId: number;
	customerDisplayName: string;
	creditType: CreditType;
	grantedAmount: number;
	termMonths: number;
	appliedAnnualInterestRate: number;
	appliedGrantingFee: number;
	customerWasVipAtCreation: boolean;
	plannedMonthlyPaymentAmount: number;
	status: CreditStatus;
	grantedAtUtc: string;
	repaidAtUtc?: string;
}
